Networking automation

TregubovYury opened the pull request:
Autor: Tregubov Yury

Added tests for QA-448 QA-449 and QA-451

Commit:
continue to work with test_543348_HeatCreateStackWaitCondition

Commit:
remove Heat_WaitCondition_543348.yaml

Commit:
revert Heat_WaitCondition_543348.yaml

Commit:
Merge remote-tracking branch 'upstream/master'

# Conflicts:
#	mos_tests/heat/heat_tests.py

Commit:
Merge remote-tracking branch 'upstream/master'

Commit:
Merge remote-tracking branch 'upstream/master'

# Conflicts:
#	mos_tests/heat/heat_tests.py

Commit:
Fit template

Commit:
Add 'test_543348_HeatCreateStackWaitCondition'

Commit:
Add template for 543349

Commit:
Fix comments about images

Commit:
change timeout for stack create

Commit:
Add 'test_543349_HeatCreateStackNeutronResources'

Commit:
Add templates for 543350

Commit:
Merge remote-tracking branch 'upstream/master'

Commit:
Add 'files' option to 'create_stack' function

Commit:
Add 'test_543350_HeatCreateStackNovaResources'

Commit:
Merge remote-tracking branch 'upstream/master'

# Conflicts:
#	mos_tests/heat/heat_tests.py

Commit:
fix pep8

Commit:
fix indent

Commit:
fix indent again

Commit:
indent!

Commit:
indent for all comments

Commit:
Merge remote-tracking branch 'upstream/master'

# Conflicts:
#	mos_tests/heat/heat_tests.py

Commit:
indent fix

Commit:
indent in comments

Commit:
indent in common functions

Commit:
indent in common functions

Commit:
Networking automation

Autor: Tregubov Yury

Added tests for QA-448 QA-449 and QA-451

Commit:
Networking automation

Autor: Tregubov Yury

Added tests for QA-448 QA-449 and QA-451

rpromyshlennikov commented on mos_tests/environment/os_actions.py:
```diff
@@ -148,6 +148,9 @@ def create_server(self, name, image_id=None, flavor=1, scenario='',
             helpers.wait(
                 lambda: self.get_instance_detail(srv).status == "ACTIVE",
                 timeout=timeout)
+            logger.info('the server {0} is {1}'.
```
Lets do it like "logger.info('the server {0} is {1}'.format(" not "logger.info('the server {0} is {1}'."

rpromyshlennikov commented on mos_tests/environment/os_actions.py:
```diff
@@ -448,3 +451,72 @@ def execute_through_host(self, ssh, vm_host, cmd, creds=()):
+            nics=[{'net-id': network_id}],
+            security_groups=[sg_id])
+
+    def reshedule_router_on_primary(self, router_id, primary_fqdn):
```
reshedule -> reschedule
check other code for this typo.

gdyuldin replied:
You should determine primary (not require it fqdn), or rename method. Also i beleive 'to' will be better than 'on' in method name

TregubovYury replied:
In this case the easiest way  is to use host's fqdn, since the l3 agents contains them.
I would propose then to name this method reschedule_router_to_primary_by_fqdn

rpromyshlennikov commented on mos_tests/environment/os_actions.py:
```diff
@@ -448,3 +451,72 @@ def execute_through_host(self, ssh, vm_host, cmd, creds=()):
+                     timeout=5 * 60)
+
+    def add_net(self, router_id):
+        i =\
```
Lets do it like i = ( 
                           ... ) 
not i =/

rpromyshlennikov commented on mos_tests/environment/os_actions.py:
```diff
@@ -448,3 +451,72 @@ def execute_through_host(self, ssh, vm_host, cmd, creds=()):
+    def add_net(self, router_id):
+        i =\
+            len(self.neutron.list_networks()['networks']) + 1
+        network =\
```
Lets do it like i = ( 
                           ... ) 
not i =/

rpromyshlennikov commented on mos_tests/environment/os_actions.py:
```diff
@@ -448,3 +451,72 @@ def execute_through_host(self, ssh, vm_host, cmd, creds=()):
+            security_groups=[sg_id])
+
+    def reshedule_router_on_primary(self, router_id, primary_fqdn):
+        agent_list =\
```
Lets do it like i = ( 
                           ... ) 
not i =/

rpromyshlennikov commented on mos_tests/environment/os_actions.py:
```diff
@@ -448,3 +451,72 @@ def execute_through_host(self, ssh, vm_host, cmd, creds=()):
+        agent_list =\
+            self.neutron.list_agents(
+                binary='neutron-l3-agent')['agents']
+        agt_id_to_move_on =\
```
Lets do it like i = ( 
                           ... ) 
not i =/

Check other code for this style.

gdyuldin commented on mos_tests/neutron/python_tests/base.py:
```diff
@@ -140,3 +140,49 @@ def check_vm_is_connectable(self, vm):
                  interval=10, timeout=3 * 10,
                  timeout_msg=error_msg)
+
+    def destroy_nodes(self, devops_nodes):
```
May be this method should be in devops client

TregubovYury replied:
At first it was planned to use both devops and fuel client in these methods.
Will check if only devops left and move them if so.

gdyuldin replied:
I believe you can separate destructive action and check action. And move them to devops client and fuel environment.

gdyuldin commented on mos_tests/neutron/python_tests/base.py:
```diff
@@ -140,3 +140,49 @@ def check_vm_is_connectable(self, vm):
+            logger.info('online state of node {0} now is {1}'.
+                         format(node.data['name'], node.data['online']))
+
+    def warm_shutdown_nodes(self, devops_nodes):
```
Same

gdyuldin commented on mos_tests/neutron/python_tests/base.py:
```diff
@@ -140,3 +140,49 @@ def check_vm_is_connectable(self, vm):
+                remote.check_call('/sbin/shutdown -Ph now')
+        self.destroy_nodes(devops_nodes)
+
+    def warm_start_nodes(self, devops_nodes):
```
same

gdyuldin commented on mos_tests/neutron/python_tests/base.py:
```diff
@@ -140,3 +140,49 @@ def check_vm_is_connectable(self, vm):
+            logger.info('online state of node {0} now is {1}'.
+                         format(node.data['name'], node.data['online']))
+
+    def warm_restart_nodes(self, devops_nodes):
```
same

gdyuldin commented on mos_tests/neutron/python_tests/base.py:
```diff
@@ -140,3 +140,49 @@ def check_vm_is_connectable(self, vm):
+        self.warm_shutdown_nodes(devops_nodes)
+        self.warm_start_nodes(devops_nodes)
+
+    def wait_nodes_get_offline_state(self, node_names=[]):
```
This should be in fuel environment class

gdyuldin commented on mos_tests/neutron/python_tests/base.py:
```diff
@@ -140,3 +140,49 @@ def check_vm_is_connectable(self, vm):
+                        if x.data['name'].split('_')[0] in node_names]
+        return all(nodes_states)
+
+    def wait_nodes_get_online_state(self):
```
same as previous
Also this method isn't wait anything

gdyuldin commented on mos_tests/neutron/python_tests/test_dhcp_agent.py:
```diff
@@ -0,0 +1,93 @@
+
+import pytest
+
+from tools.settings import logger
```
Let's not use external logger

gdyuldin commented on mos_tests/neutron/python_tests/test_dhcp_agent.py:
```diff
@@ -0,0 +1,93 @@
+        max_value = max(networks_amount_on_each_agt)
+        networks_amount_on_each_agt.remove(max_value)
+        for value in networks_amount_on_each_agt:
+            assert(self.isclose(value, max_value, abs_tol=3))
```
You can write `assert abs(max_value - value) < 3`

gdyuldin commented on mos_tests/neutron/python_tests/test_restarts.py:
```diff
@@ -0,0 +1,158 @@
+
+import pytest
+
+from tools.settings import logger
```
Let's not use external logger

gdyuldin commented on mos_tests/neutron/python_tests/test_restarts.py:
```diff
@@ -0,0 +1,158 @@
+
+    def test_shutdown_primary_controller_with_l3_agt(self):
+
+        logger.info(self.l3_agent_ids)
```
For many peoples this log record says nothing

TregubovYury replied:
yes, used for debug, just need to remove

gdyuldin commented on mos_tests/neutron/python_tests/test_restarts.py:
```diff
@@ -0,0 +1,158 @@
+        # since it will stay on the destroyed controller
+        # and remain disabled
+        self.l3_agent_ids.remove(router_agt['id'])
+        logger.info(self.l3_agent_ids)
```
For many peoples this log record says nothing

gdyuldin commented on mos_tests/neutron/python_tests/test_restarts.py:
```diff
@@ -0,0 +1,158 @@
+        # Check that the all l3 are alive
+        self.os_conn.wait_agents_alive(self.l3_agent_ids)
+
+        logger.info('agts on the router are')
```
Avoid reductions in log records

gdyuldin commented on mos_tests/neutron/python_tests/base.py:
```diff
@@ -140,3 +140,49 @@ def check_vm_is_connectable(self, vm):
+        for node in devops_nodes:
+            node.destroy()
+        logger.info('wait untill the nodes get offline state')
+        assert(wait(lambda: self.wait_nodes_get_offline_state(
```
Wait doesn't need additional assert

Commit:
Netowrking automation

Author: Tregubov Yury

changed some function names and moved several methods in Enviroment
class

Commit:
Netowrking automation

Author: Tregubov Yury

changed some function names and moved several methods in Enviroment
class

Commit:
Merge pull request #34 from AlexanderKoryagin/master

Add 'test_543348_HeatCreateStackWaitCondition', 'test_543349_HeatCreateStackNeutronResources', 'test_543350_HeatCreateStackNovaResources'

Commit:
Merge pull request #41 from TatyanaGladysheva/master

added test_543356_NovaMassivelySpawnVMsWithBootLocal

Commit:
Merge branch 'master' of https://github.com/TregubovYury/mos-integration-tests

Commit:
Networking automation

Author: Tregubov Yury

removed '=\' from the code
